java/test/jmri/jmrit/display/layoutEditor/LayoutEditorEditorTest.java

```diff
@@ -0,0 +1,348 @@
+        }
+
+        jmri.configurexml.ConfigXmlManager cm = new jmri.configurexml.ConfigXmlManager() {
+        };
+        Assert.assertNotNull("ConfigXmlManager exists", cm);
+
+        // load and display sample file
+        java.io.File leFile = new java.io.File("java/test/jmri/jmrit/display/layoutEditor/valid/SimpleLayoutEditorTest.xml");
+        cm.load(leFile);
+        sleep(100); // time for internal listeners to calm down
+
+        // Find new window by name (should be more distinctive, comes from sample file)
+        LayoutEditor le = (LayoutEditor) jmri.util.JmriJFrame.getFrame("My Layout");
+        Assert.assertNotNull("LayoutEditor exists", le);
+
+        JComponent leTargetPanel = le.getTargetPanel();
+        Assert.assertNotNull("Target Panel exists", le);
+
+        LayoutEditorFindItems leFinder = le.getFinder();
+        Assert.assertNotNull("LayoutEditorFindItems exists", leFinder);
+
+        // It's up at this point, and can be manipulated
+        // make it editable
+        le.setAllEditable(true);
+        Assert.assertTrue("isEditable after setAllEditable(true)", le.isEditable());
+
+        // setHighlightSelectedBlock(true)
+        le.setHighlightSelectedBlock(true);
+        Assert.assertTrue("getHighlightSelectedBlockafter after setHighlightSelectedBlock(true)", le.getHighlightSelectedBlock());
+
+        // find End Bumper
+        PositionablePoint ppEB1 = leFinder.findPositionablePointByName("EB1");
+        Assert.assertNotNull("End Bumper EB1 exists", ppEB1);
+
+        // Click to show popup
+        java.awt.Point location = MathUtil.PointForPoint2D(MathUtil.center(ppEB1.getBounds()));
+        getHelper().enterClickAndLeave(new MouseEventData(this,
+                leTargetPanel, // component
+                1, // number clicks
+                EventDataConstants.DEFAULT_MOUSE_MODIFIERS, // modifiers
+                true, // isPopUpTrigger
+                10, // sleeptime
+                EventDataConstants.CUSTOM, // position
+                location
+        ));
+
+        //TODO: comment out for production; 
+        // only here so developer can see what's happening
+        sleep(250);
+
+        // find last anchor
+        PositionablePoint ppA7 = leFinder.findPositionablePointByName("A7");
+        Assert.assertNotNull("Anchor A7 exists", leFinder);
+
+        // Click to show popup
+        location = MathUtil.PointForPoint2D(MathUtil.center(ppA7.getBounds()));
+        getHelper().enterClickAndLeave(new MouseEventData(this,
+                leTargetPanel, // component
+                1, // number clicks
+                EventDataConstants.DEFAULT_MOUSE_MODIFIERS, // modifiers
+                true, // isPopUpTrigger
+                10, // sleeptime
+                EventDataConstants.CUSTOM, // position
+                location
+        ));
+
+        //TODO: comment out for production; 
+        // only here so developer can see what's happening
+        sleep(250);
```

Comment:
You say you are going to comment this out for production, I would remove it completely.

Reply:
I find that I often end up putting these types of things back in when I need to diagnose a problem in the future… and it's a lot easier to uncomment something than re-enter it.

Reply:
They clutter the code, and serve no real purpose.  We generally don't like leaving code in place that isn't used.